JAVA4/src/solution/KYCDateRange.java

```diff
@@ -0,0 +1,78 @@
+package solution;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+/**
+ * Carries out all calculation for range of form date filling.
+ */
+public class KYCDateRange{
+    private final SimpleDateFormat dateform =new SimpleDateFormat("dd-MM-yyyy");
+    private Date signup;
+    private Date current;
+    /**
+     * Parameterised constructor
+     * @param signupdate signup date of customer
+     * @param currdate the current date when customer if filling the form
+     */
+    public KYCDateRange(String signupdate,String currdate)  {
```

Comment:
In this case, your program can not work without valid dates. So instead of handling the exception, you have to throw the exception to caller.